packages/grid/x-data-grid/src/constants/envConstants.ts

```diff
@@ -22,13 +22,13 @@ let experimentalEnabled = false;
 
 if (
   typeof process !== 'undefined' &&
-  process.env.GRID_EXPERIMENTAL_ENABLED !== undefined &&
+  process.env?.GRID_EXPERIMENTAL_ENABLED !== undefined &&
```

Comment:
Thank you for your contribution!

What do you think about making in the initial check more explicit?
```ts
process?.env &&
process.env.GRID_EXPERIMENTAL_ENABLED !== undefined &&
```
The same idea/suggestion would apply to the L30 & L31.

Reply:
@LukasTy We could, but essentially these are the same -

`process?.env?.Foo`
`process && process.env && process.env.Foo`.

Reply:
Yes, they are, my idea was that in this case the existing check for `process` is redundant as it is defined in your runtime as well, but what is undefined is actually `env` object.
That's why I was suggesting to replace the existing check to actually safe-guard against environments where `env` object is not exposed/loaded.